homeassistant/components/lyric/climate.py

```diff
@@ -281,23 +291,61 @@ async def async_set_temperature(self, **kwargs) -> None:
             try:
                 if self.hvac_mode == HVAC_MODE_COOL:
                     await self._update_thermostat(
-                        self.location, device, coolSetpoint=temp
+                        self.location,
+                        device,
+                        coolSetpoint=temp
                     )
                 else:
                     await self._update_thermostat(
-                        self.location, device, heatSetpoint=temp
+                        self.location,
+                        device,
+                        heatSetpoint=temp
                     )
             except LYRIC_EXCEPTIONS as exception:
                 _LOGGER.error(exception)
-        await self.coordinator.async_refresh()
+            await self.coordinator.async_refresh()
 
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set hvac mode."""
-        _LOGGER.debug("Set hvac mode: %s", hvac_mode)
+        _LOGGER.debug("HVAC mode from frontend: %s", hvac_mode)
```

Comment:
The frontend is not the only thing that can call services.
```suggestion
        _LOGGER.debug("HVAC mode: %s", hvac_mode)
```